Flaky test: migration/test_to25.py::test_that_migration_creates_experiment_index_with_params_responses_and_obs_configs

Sorting is apparently not sufficiently determistic for this test.

```
FAILED tests/ert/unit_tests/storage/migration/test_to25.py::test_that_migration_creates_experiment_index_with_params_responses_and_obs_configs - AssertionError: assert {'ensembles':...t-experiment'} == {'ensembles':...t-experiment'}
  
  Omitting 3 identical items, use -vv to show
  Differing items:
  {'experiment': {'observations': [{'date': '2020-01-01', 'error': 0.05, 'key': 'FOPR', 'name': 'FOPR', ...}, {'data': '...': 'summary'}, {'keys': ['SOME', 'OTHER', 'OVERLAP'], 'name': 'gen_data', 'type': 'gen_data'}], 'weights': [0.2, 0.8]}} != {'experiment': {'observations': [{'data': 'SOME', 'error': 0.1, 'index': 0, 'name': 'OBS1', ...}, {'data': 'SOME', 'er...': 'summary'}, {'keys': ['SOME', 'OTHER', 'OVERLAP'], 'name': 'gen_data', 'type': 'gen_data'}], 'weights': [0.2, 0.8]}}
  
  Full diff:
    {
        'ensembles': [],
        'experiment': {
            'observations': [
  +             {
  +                 'date': '2020-01-01',
  +                 'error': 0.05,
  +                 'key': 'FOPR',
  +                 'name': 'FOPR',
  +                 'type': 'summary_observation',
  +                 'value': 3.21,
  +             },
                {
                    'data': 'SOME',
                    'error': 0.1,
                    'index': 0,
                    'name': 'OBS1',
                    'restart': 1,
                    'type': 'general_observation',
                    'value': 1.23,
                },
                {
                    'data': 'SOME',
                    'error': 0.2,
                    'index': 1,
                    'name': 'OBS2',
                    'restart': 2,
                    'type': 'general_observation',
                    'value': 2.34,
  -             },
  -             {
  -                 'date': '2020-01-01',
  -                 'error': 0.05,
  -                 'key': 'FOPR',
  -                 'name': 'FOPR',
  -                 'type': 'summary_observation',
  -                 'value': 3.21,
                },
            ],
```

--- a/src/ert/storage/migration/to25.py
+++ b/src/ert/storage/migration/to25.py
@@ -49,7 +49,7 @@ def _observation_row_to_declaration(
 def migrate_parameters_responses_and_observations_into_experiment_index(
     path: Path,
 ) -> None:
-    for exp_path in path.glob("experiments/*"):
+    for exp_path in sorted(path.glob("experiments/*")):
         if not exp_path.is_dir():
             continue
 
@@ -67,7 +67,7 @@ def migrate_parameters_responses_and_observations_into_experiment_index(
 
         obs_dir = exp_path / "observations"
         if obs_dir.exists():
-            for obs_file in obs_dir.glob("*"):
+            for obs_file in sorted(obs_dir.glob("*")):
                 df = pl.read_parquet(obs_file)
                 experiment_json["observations"].extend(
                     [

